extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/TestPortUtil.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.fsinternetradio.test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+/**
+ * Look up a free TCP/IP port on localhost for test execution. This ensures collision free parallel test execution using
+ * embedded jetty on dynamic ports.
+ *
+ * @author Henning Treu - initial contribution
+ *
+ */
+public class TestPortUtil {
+
+    /**
+     * Returns a free port number on localhost.
+     *
+     * Heavily inspired from org.eclipse.jdt.launching.SocketUtil (to avoid a dependency to JDT just because of this).
+     * Slightly improved with close() missing in JDT. And throws exception instead of returning -1.
+     *
+     * @return a free port number on localhost
+     * @throws IllegalStateException if unable to find a free port
+     */
+    public static int findFreePort() {
+        ServerSocket socket = null;
+        try {
+            socket = new ServerSocket(0);
+            socket.setReuseAddress(true);
+            int port = socket.getLocalPort();
+            try {
+                socket.close();
+            } catch (IOException e) {
+                // Ignore IOException on close()
+            }
+            return port;
+        } catch (IOException e) {
+        } finally {
+            if (socket != null) {
+                try {
+                    socket.close();
+                } catch (IOException e) {
+                }
+            }
+        }
+        throw new IllegalStateException("Could not find a free TCP/IP port to start embedded Jetty HTTP Server on");
```

Comment:
The only reason to not return the port is if above (opening socket, not the finally branch) the `IOException` is thrown.
So move this method to the catch branch and embed the `IOException` into the ISE.